### DISCUSSION TITLE
Error in Blazor App When Upgrading from 5.4.1 to 5.4.2 or Higher

### DISCUSSION THREAD

**Discussion by saulyt** [MarimerLLC/csla#2600] [Questions] [answered]
I tried upgrading my business library and Blazor application from version 5.4.1 to 5.5.1 and got the the following error in both a Blazor Server and Blazor Webassembly version of the application:
`InvalidOperationException: The AuthorizationPolicy named: 'Csla:GetObject|NameofMyBusineesClass, Version=VersionOfMyLibrary, Culture=neutral, PublicKeyToken=null' was not found.`

I got the error for each item in my NavMenu that is using a CSLA authorization rule for the policy :

`<AuthorizeView Policy="@(CslaPolicy.GetPolicy(Csla.Rules.AuthorizationActions.GetObject, typeof(NameofMyBusineesClass)))">
	<li class="nav-item">
		<NavLink class="nav-link text-dark" href="mypage">
			<span class="fa fa-gavel" aria-hidden="true"></span> My Page
		</NavLink>
	</li>
</AuthorizeView>
`
I tried downgrading to 5.4.2 and still got this error. Reverting to 5.4.1 fixed the issue. Is there a known breaking change that was made from 5.4.1 to 5.4.2 that would cause this? I tried [comparing](https://github.com/MarimerLLC/csla/compare/v5.4.1...v5.4.2) the 2 releases but nothing stands out to me that would cause this.

Thank you.

**Comment by rockfordlhotka** [comment, marked as answer]
Not that I know of. There are a couple types that need to be registered with DI on startup to support CSLA claims - I wonder if they aren't being registered somehow?

```c#
        services.AddTransient<IAuthorizationPolicyProvider, CslaPermissionsPolicyProvider>();
        services.AddTransient<IAuthorizationHandler, CslaPermissionsHandler>();
```

**Reply by saulyt** [reply]
Thanks Rocky! That fixed it. I don't understand why it worked with 5.4.1 without those 2 registrations but I'll take it.
